src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -62,15 +70,31 @@ public void locateComponentsForOnlineIntelligentScan() throws ComponentLocatorEx
     /**
      * A ComponentLocatorException should be thrown in all cases where the output file is not created so that the
      * appropriate status can be logged for the requested {@link GenerateComponentLocationAnalysisOperation}.
+     * @throws com.synopsys.integration.detect.workflow.componentlocationanalysis.ComponentLocatorException
      */
     public void failComponentLocationAnalysisOperation() throws ComponentLocatorException {
         throw new ComponentLocatorException("Failed to generate Component Location Analysis file.");
     }
+    
+    private File serializeInputToJson(File saveInputFileDir, Input libInput) throws DetectUserFriendlyException {
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        String serializedLibInput = gson.toJson(libInput);
+        try {
+            File componentsSourceInputFile =  new File (saveInputFileDir, LOCATOR_INPUT_FILE_NAME);
+            DetectFileUtils.writeToFile(componentsSourceInputFile, serializedLibInput);
+            logger.debug("Component Location Analysis input file written to {}", componentsSourceInputFile);
+            return componentsSourceInputFile;
+        } catch (IOException ex) {
+            throw new DetectUserFriendlyException("Failed to create component location analysis output file", ex, ExitCodeType.FAILURE_UNKNOWN_ERROR);
```

Comment:
Do you want the Detect run to fail if something goes wrong with serializing this file?